plugin/src/main/java/org/wildfly/plugin/provision/GlowConfig.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.plugin.provision;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.wildfly.glow.Arguments;
+import org.wildfly.glow.OutputFormat;
+import org.wildfly.glow.ScanArguments.Builder;
+
+/**
+ *
+ * @author jdenise
+ */
+public class GlowConfig {
+
+    String context = "bare-metal";
+    String profile;
+    Set<String> addOns = Collections.emptySet();
+    String version;
+    boolean suggest;
+    private Set<String> layersForJndi = Collections.emptySet();
+    private boolean failsOnError = true;
+    boolean preview;
+
+    public GlowConfig() {
+    }
+
+    public Arguments toArguments(Path deployment, Path inProvisioning) {
+        Set<String> profiles = new HashSet<>();
+        if (profile != null) {
+            profiles.add(profile);
+        }
+        List<Path> lst = new ArrayList<>();
+        lst.add(deployment);
```

Comment:
If this is the only entry, we should just use `List.of(deployment);`.

Reply:
Ok, thanks.